File: client/Research2DPrototype/Assets/Scripts/SI/PowerValue.cs
using System;

public class PowerValue
{
  public double Value { get; private set; }
  public int Power { get; private set; }

  private double _calculatedValue;
  public double CalculatedValue
  {
    get
    {
      if (this.Power == 1)
        return _calculatedValue;
      else
        return _calculatedValue * Math.Pow(10, this.Power);
    }
  }

  // public PowerValue() {}

  public PowerValue(double v, int p)
  {
    this.Value = v;
    this.Power = p;
  }  
}

File: shared_data/power_value.rb

class PowerValue
    # 6.62607015×10^−34 is (6.62607015, -34)
    def initialize(value, power)
        @value = value # float
        @power = power # int
    end

    def calculate
        if @power == 1
            return @value
        else
            return @value * 10 ** @power
    end
end
